### PR TITLE
Add Swift Package Manager (SPM) support

### PR DESCRIPTION
**Pull request by david-swift**
Hi!

This PR adds support for [SPM](https://www.swift.org/documentation/package-manager/) (e.g. useful in combination with the [Swift 5 SDK Extension](https://github.com/flathub/flathub/pull/4935)). It contains a script for generating the necessary  files, a README and a sample Swift package.

### CODE REVIEW

File: spm/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "buildsystem": "simple",
+    "sources": [
+        {
+            "type": "dir",
+            "path": "."
+        },
+        "generated-sources.json"
+    ],
+    "build-commands": [
+        "chmod +x ./setup-offline.sh ./setup-offline.sh",
```
Comment (TingPing) [inline spm/README.md:32]: Just make it executable at time of creation?

File: spm/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        "./setup-offline.sh",
+        "swift build -c release --static-swift-stdlib",
+        "rm -r -f /app/bin",
```
Comment (TingPing) [inline spm/README.md:35]: Why is this necessary?

**Review by TingPing — COMMENTED** [review]
Thanks for the contribution, Swift support would be great!

Are you going to commit to maintaining this here? As-in, is it OK to set you as a codeowner so new issues are assigned to you.

File: spm/README.md (hunk 2 above)
Comment (david-swift) [reply spm/README.md:34]: It actually isn't...

**Comment by david-swift** [comment]
> Thanks for the contribution, Swift support would be great!
> 
> Are you going to commit to maintaining this here? As-in, is it OK to set you as a codeowner so new issues are assigned to you.

@TingPing  Yes, that's ok. Thanks for the review!

File: spm/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            "type": "dir",
+            "path": "."
+        },
+        "generated-sources.json"
+    ],
+    "build-commands": [
+        "./setup-offline.sh",
+        "swift build -c release --static-swift-stdlib",
+        "mkdir /app/bin",
+        "cp .build/release/quickstart /app/bin"
```
Comment (TingPing) [inline spm/README.md:35]:
```suggestion
        "install -Dm755 .build/release/quickstart /app/bin/quickstart",
```

File: spm/quickstart/Sources/quickstart/main.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// The Swift Programming Language
+// https://docs.swift.org/swift-book
+
+import Adwaita
+
+@main
+struct QuickStart: App {
+
+    let id = "com.flatpak.quickstart"
```
Comment (hfiguiere) [inline spm/quickstart/Sources/quickstart/main.swift:9]:
```suggestion
    let id = "org.flatpak.quickstart"
```

File: spm/quickstart/com.flatpak.quickstart.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{
+    "app-id": "com.flatpak.quickstart",
```
Comment (hfiguiere) [inline spm/quickstart/com.flatpak.quickstart.json:2]:
```suggestion
    "app-id": "org.flatpak.quickstart",
```

and the file should be renamed.

I know it's just an example, but at least it should conform to the requirements.

**Comment by hfiguiere** [comment]
My biggest issue here is that the tool is in Swift. Nothing against Swift but it makes it more difficult to run and to maintain when everything else is in Python.

**Comment by david-swift** [comment]
> My biggest issue here is that the tool is in Swift. Nothing against Swift but it makes it more difficult to run and to maintain when everything else is in Python.

Rewriting in Python wouldn't be too difficult, but the generation itself relies on Swift being installed. Isn't the idea that the app maintainers run those tools themselves? Installing Swift in a GitHub action for example is really simple, and people using this tool probably have Swift installed on their machine. Or do I misunderstand something, @hfiguiere?

File: spm/quickstart/com.flatpak.quickstart.json (hunk 5 above)
Comment (hfiguiere) [reply spm/quickstart/com.flatpak.quickstart.json]: oh, I should fix that one then.

**Comment by TingPing** [comment]
I do agree being in Python would be a little easier to maintain and use.

However I'm fine merging this as-is. You're putting in the work and if Swift works best for you then that's fine.

**Comment by hfiguiere** [comment]
fine by me too. I just wanted to note the caveats
